### PR TITLE
[Enhancement]Use run mode to replace use_staros and integrate_starmgr

### PR DESCRIPTION
**Pull request by abc982627271**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x]  3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 26 / 36 (72.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|0|1|00.00%|[364]|
|:large_blue_circle:|com/starrocks/catalog/ColocateTableIndex.java|0|1|00.00%|[1012]|
|:large_blue_circle:|com/starrocks/StarRocksFE.java|0|1|00.00%|[134]|
|:large_blue_circle:|com/starrocks/system/ComputeNode.java|1|2|50.00%|[466]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|2|4|50.00%|[88, 89]|
|:large_blue_circle:|com/starrocks/sql/common/EngineType.java|2|3|66.67%|[34]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|10|13|76.92%|[661, 666, 667]|
|:large_blue_circle:|com/starrocks/common/proc/BackendsProcDir.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/RunMode.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|1|1|100.00%|[]|

**Comment by abc982627271** [comment]
run  starrocks_fe_unittest

**Review by sduzh — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/ColocateTableIndex.java [hunk 1]
```diff
@@ -1011,7 +1009,7 @@ public void updateLakeTableColocationInfo(OlapTable olapTable) throws DdlExcepti
     }
 
     private void constructLakeGroups(GlobalStateMgr globalStateMgr) {
-        if (!Config.use_staros) {
+        if (GlobalStateMgr.getCurrentState().isSharedNothingMode()) {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/catalog/ColocateTableIndex.java:1012]: use `!GlobalStateMgr.getCurrentState().isSharedDataMode()`

currently `GlobalStateMgr.getCurrentState().isSharedNothingMode()` is equivalent to `!GlobalStateMgr.getCurrentState().isSharedDataMode()`, when a third mode is introduced, the assumption doesn't hold any more.

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java [hunk 2]
```diff
@@ -82,17 +82,11 @@ public StarOSAgent() {
...
-                LOG.error("integrate_starmgr is true but use_staros is false!");
-                return false;
-            }
-            // check if Config.starmanager_address == FE address
-            String[] starMgrAddr = Config.starmgr_address.split(":");
-            if (!starMgrAddr[0].equals("127.0.0.1")) {
-                LOG.error("Config.starmgr_address not equal 127.0.0.1, it is {}", starMgrAddr[0]);
-                return false;
-            }
+        // check if Config.starmanager_address == FE address
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:85]: shall check current mode is sharednothing mode?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java [hunk 3]
```diff
@@ -76,14 +75,6 @@ private static String analyzeEngineName(String engineName) {
...
         }
 
-        if (engineName.equalsIgnoreCase(EngineType.STARROCKS.name()) && !Config.use_staros) {
-            throw new SemanticException("Engine %s needs 'use_staros = true' config in fe.conf", engineName);
-        }
-
-        if (engineName.equalsIgnoreCase(EngineType.OLAP.name()) && Config.use_staros) {
-            throw new SemanticException("Disallow create OLAP table in this cluster");
-        }
-
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java:86]: whether need to keep this check?

reject create olap table in shared-data mode, reject create starrocks table in shared-nothing mode?

Comment (abc982627271) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java:86]: not need, if user create olap table in shared-data mode, we will create a lake table for him.

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java (hunk 2 above)
Comment (abc982627271) [reply fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:85]: No need.
image
 Already checked before call init() of StarOSAgent.java

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java (hunk 3 above)
Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java:86]: but the engine will still show as olap? should not do this until starrocks engine is completely removed.

Comment (abc982627271) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java:86]: yes,  remove starrocks engine in this pr.

Comment (abc982627271) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java:86]: Will remove starrrocks engine in this pr: https://github.com/StarRocks/starrocks/pull/18122/files

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java [hunk 4]
```diff
@@ -76,11 +76,14 @@ private static String analyzeEngineName(String engineName) {
             return EngineType.defaultEngine().name();
         }
 
-        if (engineName.equalsIgnoreCase(EngineType.STARROCKS.name()) && !Config.use_staros) {
-            throw new SemanticException("Engine %s needs 'use_staros = true' config in fe.conf", engineName);
+
+        if (engineName.equalsIgnoreCase(EngineType.STARROCKS.name()) &&
+                !GlobalStateMgr.getCurrentState().isSharedDataMode()) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java:81]:
```suggestion
                GlobalStateMgr.getCurrentState().isSharedNothingMode()) {
```

Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java:81]:
```suggestion
                GlobalStateMgr.getCurrentState().isSharedNothingMode()) {
```

Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java:81]: lake table is allowed in hybrid mode

**Review by wyb — DISMISSED** [review]

**Review by tracymacding — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by sduzh — DISMISSED** [review]

**Review by kevincai — DISMISSED** [review]

**Comment by wangruin** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 30 / 39 (76.92%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|0|1|00.00%|[364]|
|:large_blue_circle:|com/starrocks/catalog/ColocateTableIndex.java|0|1|00.00%|[1012]|
|:large_blue_circle:|com/starrocks/StarRocksFE.java|0|1|00.00%|[134]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|2|4|50.00%|[88, 89]|
|:large_blue_circle:|com/starrocks/sql/common/EngineType.java|2|3|66.67%|[34]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableAnalyzer.java|4|5|80.00%|[81]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|9|11|81.82%|[670, 674]|
|:large_blue_circle:|com/starrocks/common/proc/BackendsProcDir.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/RunMode.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/system/ComputeNode.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|1|1|100.00%|[]|

**Review by sduzh — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by wyb — APPROVED** [review]

**Comment by abc982627271** [comment]
@mergifyio backport branch-3.0

**Pull request merged**